Marlin/ConfigurationStore.cpp

```diff
@@ -1,33 +1,92 @@
+/**
+ * ConfigurationStore.cpp
+ *
+ * Configuration and EEPROM storage
+ *
+ * V15 EEPROM Layout:
+ *
+ *  ver
+ *  axis_steps_per_unit (x4)
+ *  max_feedrate (x4)
+ *  max_acceleration_units_per_sq_second (x4)
+ *  acceleration
+ *  retract_acceleration
+ *  minimumfeedrate
+ *  mintravelfeedrate
+ *  minsegmenttime
+ *  max_xy_jerk
+ *  max_z_jerk
+ *  max_e_jerk
+ *  add_homing (x3)
+ *
+ * DELTA:
+ *  endstop_adj (x3)
+ *  delta_radius
+ *  delta_diagonal_rod
+ *  delta_segments_per_second
+ *
+ * ULTIPANEL:
+ *  plaPreheatHotendTemp
+ *  plaPreheatHPBTemp
+ *  plaPreheatFanSpeed
+ *  absPreheatHotendTemp
+ *  absPreheatHPBTemp
+ *  absPreheatFanSpeed
+ *  zprobe_zoffset
+ *
+ * PIDTEMP:
+ *  Kp[0], Ki[0], Kd[0], Kc[0]
+ *  Kp[1], Ki[1], Kd[1], Kc[1]
+ *  Kp[2], Ki[2], Kd[2], Kc[2]
+ *  Kp[3], Ki[3], Kd[3], Kc[3]
+ *
+ * DOGLCD:
+ *  lcd_contrast
+ *
+ * SCARA:
+ *  axis_scaling (x3)
+ *
+ * FWRETRACT:
+ *  autoretract_enabled
+ *  retract_length
+ *  retract_length_swap
+ *  retract_feedrate
+ *  retract_zlift
+ *  retract_recover_length
+ *  retract_recover_length_swap
+ *  retract_recover_feedrate
+ *
+ *  volumetric_enabled
+ *
+ *  filament_size (x4)
+ *
+ */
 #include "Marlin.h"
 #include "planner.h"
 #include "temperature.h"
 #include "ultralcd.h"
 #include "ConfigurationStore.h"
 
-void _EEPROM_writeData(int &pos, uint8_t* value, uint8_t size)
-{
-    do
-    {
-        eeprom_write_byte((unsigned char*)pos, *value);
-        pos++;
-        value++;
-    }while(--size);
+void _EEPROM_writeData(int &pos, uint8_t* value, uint8_t size) {
+  do {
```

Comment:
@thinkyhead 
Hi Scott,
great work! This should help a lot.
What do you think, wouldn't it be a good idea to implement a write verify here? 
We could read the written *value and check if it worked. If not, give it some 2 or 3 retries.
If the retries don't work, give a error message to serial.
Something like
```CPP
void _EEPROM_writeData(int &pos, uint8_t* value, uint8_t size) {
  uint8_t retries,
              value_read;
  int pos_read;

  do {

    retries = 0;
    do {
      eeprom_write_byte((unsigned char*)pos, *value);
      value_read = eeprom_read_byte((unsigned char*)pos);
    } while ( (*value != value_read) && (retries++<3) )

    if (*value != value_read) {
      SERIAL_ECHO_START;
      SERIAL_ECHOPAIR("ERROR writing to EEPORM: at pos ", pos);
      SERIAL_ECHOPAIR(" wrote " , *value);
      SERIAL_ECHOPAIR(" read " , value_read);
      SERIAL_ECHOLN("");
    }

    pos++;
    value++;
  } while (--size);
}
```
This is just a q+d hack, but could help.

This would perhaps fix #416 in nearly most cases. @daid told 20 days ago :
> We had to replace a few Arduino's at Ultimaker due to the EEPROM not working properly (settings got lost all the time)

In all other cases there would be a feedback to the user.

Cheers